### DISCUSSION TITLE
Is there a way to insert Movement data collected as digital data?

### DISCUSSION THREAD

**Discussion by GuilleSGCUPO** [petersenpeter/CellExplorer#97] [Q&A] [answered]
Hi guys, we are trying to add a digital event channel (with movement data) to the CellExplorer pipeline in order to determine differences in the firing rate of the neurons based on it. We are trying to do it with the Behavioral tracking but we do not get a proper result as it works with TTL data. Do you know any other way to make it possible? Any help would be welcome :)
Thanks for your help with other discussions.

**Comment by petersenpeter** [comment]
Hi
This is possible with the behavior struct:https://cellexplorer.org/datastructure/data-structure-and-format/#behavior

You want to save your TTL timestamps in the timestamps field and your x,y,z data in the position fields:
```m
behaviorName.timestamps % Your TTL timestamps
behaviorName.position.x % Your x position data.
behaviorName.position.y % Your y position data.
behaviorName.position.z % Your z position data if available
```

There is an example dataset you can use to better understand the data structure: 
Data path:
https://buzsakilab.nyumc.org/datasets/PetersenP/CellExplorerExampleData/MS22/Peter_MS22_180629_110319_concat/
Behavior struct: 
https://buzsakilab.nyumc.org/datasets/PetersenP/CellExplorerExampleData/MS22/Peter_MS22_180629_110319_concat/Peter_MS22_180629_110319_concat.circular_track.behavior.mat

**Reply by GuilleSGCUPO** [reply]
Thanks so much Peter! I am trying to do it following the "manipulations" part of the tutorial because I want to exclude that movement data. However, we have tried to follow the behavioral example "Peter_MS22_180629_110319_concat.SleepState.states" in order to compare both states inside the CellExplorer gui, though it did not work out yet.

We have no position data, only the event times when the animal moves. I do not know if this is enough to use the behavioral tracking tutorial.

Thanks again :)

**Comment by petersenpeter** [comment, marked as answer]
OK, in that case, you can either define a manipulation struct with intervals with movement or provide the intervals as an input when you run ProcessCellMetrics: https://cellexplorer.org/tutorials/manipulation-tutorial/#example-2-excluding-manipulations-for-waveform-metrics-by-supplying-intervals

```
excludeIntervals = [0,2000;20010,2020]; % in seconds - pairs of intervals
% Now compute the metrics while using the input parameter: excludeIntervals
cell_metrics_control  = ProcessCellMetrics('session', session,'saveAs','cell_metrics_control','excludeIntervals',excludeIntervals);
```
In the above code example, I save the metrics with a different name, but you can also run the pipeline without saving the intervals.

**Reply by GuilleSGCUPO** [reply]
Thanks so much Peter, I did it by defining the Movement intervals and it works perfectly. Besides it shows the time-points where the animal moved. So useful for our work :)
Thanks again
